### DISCUSSION TITLE
hoiw do I select the whole record/all columns?

### DISCUSSION THREAD

**Discussion by Maoie86** [Intechnity-com/OdooJsonRpcClient#75] [Q&A]
I see sample to query a model using the ff:

var products = await repository.Query()
.Where(x => x.Name, OdooOperator.EqualsTo, "test product name")
.Where(x => x.WriteDate, OdooOperator.GreaterThanOrEqualTo, new DateTime(2020, 12, 2))
.Select(x => new
{
x.Name,
x.Description,
x.WriteDate
})
.OrderByDescending(x => x.Id)
.Take(10)
.ToListAsync();

How do i select all fields/colums or the whole record?

TIA,

Mu

**Comment by patricoos** [comment]
Remove 
```
.Select(x => new
{
x.Name,
x.Description,
x.WriteDate
})
```

**Reply by Maoie86** [reply]
Thanks for the reply.   This is my new query:

                var records = await MyGlobals.SaleOrderRepo.Query()
                    .Where(x => x.DateOrder, OdooOperator.GreaterThanOrEqualTo, new DateTime(2020, 12, 2))
                    .OrderBy(x => x.Id)
                    .ToListAsync();

I got this error:

System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null. (Parameter 'source')
  Source=System.Linq
  StackTrace:
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
   at SFIOdooApp.MainV.<CopySaleOrderOdoo2MDB>d__13.MoveNext() in E:\vs2022\SFIOdooApp\SFIOdooApp\MainV.cs:line 304

**Reply by Maoie86** [reply]
why thumbsdown?  any solution to this error?
